__tests__/app.test.js

```diff
@@ -105,29 +105,159 @@ beforeEach(() => {
+  })
 
+  
 
-it('Response 400 and appropriate message if article_id is not a number', () => {
-    return request(app)
-        .get('/api/articles/banana')
-        .expect(400)
-        .then((response) => {
-          let articles = response.body.msg
+  describe('api/articles/:articleId/comments', () => {
 
-            expect(articles).toEqual('Invalid ID')
-        })
-})
+    it('GET /:article_id with comments for that id  return 200 and an object containing article key, an array of article with comments', () => {
+      
+      return request(app)
+        .get('/api/articles/1/comments')
+          .expect(200)
+            .then((response) => {
 
+      let articles = response.body.comment
+
+
+          expect(articles).toBeInstanceOf(Array);
+          expect(articles.length).toBe(11);
+         
+          articles.forEach((comment) => {
+            expect(comment).toEqual(
+              expect.objectContaining({
+              
+                article_id: expect.any(Number),
+                comment_id: expect.any(Number),
+body:expect.any(String),
+                author: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number)
+              }))
+            })
+            
+            })
+          })
+
+          test('Response 400 and appropriate message if article_id is a number but article does not exist', () => {
+            return request(app)
+                .get('/api/articles/10000000')
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.message).toEqual('Bad Request')
+                })
+          })
+          
+          test('Response 400 and appropriate message if article_id is not a number', () => {
+            return request(app)
+                .get('/api/articles/banana')
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.message).toEqual('Bad Request')
+                })
+          })
+          
+        
   })
 
   
+
+  describe('POST /api/articles/:article_id/comments', () => {
+    it.only('Response 201 and returns newly inserted comment in an object with key of comment', () => {
+        const newComment = {
+            username: `butter_bridge`,
+            body: `Posting a new comment`
+        }
+        return request(app)
+            .post('/api/articles/2/comments')
+            .send(newComment)
+            .expect(201)
+            .then(({ body }) => {
+                expect(body.comment[0]).toEqual({
+                    article_id: 2,
+                    author: "butter_bridge",
+                    body: "Posting a new comment",
+                    comment_id: 19,
+                    created_at: expect.any(String),
+                    votes: 0,
+                })
+            })
+    })
+
+    it('should respond with a status 400 Bad request', ()=>{
+      return request(app)
+      .post('/api/articles/15/comments')
```

Comment:
Line 208, I can't see what makes this a bad request? Unclear what this is testing for

Reply:
A 400 bad request example would be /api/articles/banana/comments